ocaml/xapi/message_forwarding.ml

```diff
@@ -5287,6 +5287,36 @@ functor
         info "SDN_controller.forget: sdn_controller = '%s'"
           (sdn_controller_uuid ~__context self) ;
         Local.SDN_controller.forget ~__context ~self
+
+      let add_local_port ~__context ~self ~protocol ~port =
+        info
+          "SDN_controller.add_local_port: sdn_controller = '%s', protocol = '%s', \
+           port = '%s'"
+          (sdn_controller_uuid ~__context self)
+          (Record_util.sdn_port_protocol_to_string protocol)
+          (Int64.to_string port) ;
+        let local_fn = Local.SDN_controller.add_local_port ~self ~protocol ~port in
+        List.iter
+          (fun host ->
+            do_op_on ~local_fn ~__context ~host (fun session_id rpc ->
+                Client.SDN_controller.add_local_port rpc session_id self protocol
+                  port))
+          (Db.Host.get_all ~__context)
```

Comment:
I think this part is no longer required since the port opening/closing is hanfled by `openvswitch-config-update`

Reply:
If the operation is just a DB update, then you don't need any forwarding, and just execute the `Local` functions. However, think about whether there is a need to do "live" updates or not.